pkg/apis/kudo/v1beta1/instance_types_test.go

```diff
@@ -74,3 +75,27 @@ func TestGetLastExecutedPlanStatus(t *testing.T) {
 		}
 	}
 }
+
+func TestSpecParameterDifference(t *testing.T) {
+	var testParams = []struct {
+		name string
+		new  map[string]string
+		diff map[string]string
+	}{
+		{"update one value", map[string]string{"one": "11", "two": "2"}, map[string]string{"one": "11"}},
+		{"update multiple values", map[string]string{"one": "11", "two": "22"}, map[string]string{"one": "11", "two": "22"}},
+		{"add new value", map[string]string{"one": "1", "two": "2", "three": "3"}, map[string]string{"three": "3"}},
+		{"remove one value", map[string]string{"one": "1"}, map[string]string{"two": "2"}},
+		{"no difference", map[string]string{"one": "1", "two": "2"}, map[string]string{}},
+		{"empty new map", map[string]string{}, map[string]string{"one": "1", "two": "2"}},
+	}
+
+	g := gomega.NewGomegaWithT(t)
+
+	var old = map[string]string{"one": "1", "two": "2"}
+
+	for _, test := range testParams {
+		diff := parameterDiff(old, test.new)
+		g.Expect(diff).Should(gomega.Equal(test.diff), test.name)
```

Comment:
Not yours, but since this is the only use of `gomega` in our tests, we could also get rid of that dependency and instead do
```go
if !reflect.DeepEqual(diff, test.diff) {
  t.Errorf("%s: Excepted %v, got %v", test.name, diff, test.diff)
}
```

Reply:
Good eye!